### DISCUSSION TITLE
New behavior in cargo.toml: adding minor versions

### DISCUSSION THREAD

**Discussion by yanns** [renovatebot/renovate#28782] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We are using cargo for rust projects.
For dependencies that we want to update once per week, we don't specify any minor version, like:
```
some_crate = "1.5"
``` 
The minor version is in the `Cargo.lock` file, for example `1.5.4`.

We ask renovate to update the `Cargo.lock` file so that we can update all minor dependencies once a week in one PR.
That updates does not change `Cargo.toml`.

Recently, renovate opened lots of PR adding the minor version in `Cargo.toml`, like:
```
- some_crate = "1.5"
+ some_crate = "1.5.4"
``` 
This change does not have any impact on the application as the version `1.5.4` is already being used in `Cargo.lock`.

Those kind of changes creates lots of noise. And we would like to avoid them.

Either there are some recent changes that "broke" the previous behavior that was perfectly fine for us.
Or we might need to set a configuration flag to avoid those kind of PRs?

Any help appreciated.

**Comment by viceice** [comment]
expected behavior:
- https://github.com/renovatebot/renovate/pull/28632
- https://github.com/renovatebot/renovate/pull/28714

**Reply by yanns** [reply]
So once https://github.com/renovatebot/renovate/pull/28714 is merged, we should not see all those unnecessary PR updates anymore right?

**Reply by viceice** [reply]
yes

**Reply by yanns** [reply]
thanks a lot for the quick clarification! :heart:

**Reply by rarkins** [reply, marked as answer]
I've decided to backport this to v37: https://github.com/renovatebot/renovate/pull/28826

@nabeelsaabna can you deploy today?

**Reply by nabeelsaabna** [reply]
done
updated to latest `37.340.10`

**Comment by yanns** [comment]
I'm still observing the same behavior with mend renovate:
![image](https://github.com/renovatebot/renovate/assets/51669/84609866-9ae7-4ebd-81fd-c7a34abd6ce2)

![image](https://github.com/renovatebot/renovate/assets/51669/25bbae3f-3b1f-4b7c-b5ac-e8662048afa0)

The project does not specify any minor version in `Cargo.toml`:
```
[dependencies]
parking_lot = { version = "0.12", features = ["hardware-lock-elision"] }
```

**Reply by rarkins** [reply]
It's a lock file update and looks to be exactly as we intend it. What is confusing or concerning to you about it?

**Reply by rarkins** [reply]
Also this does not sound like "same behavior" as you originally described, because you originally described behavior where the toml file was unnecessarily changed. Please clarify why you consider it "same" behavior if I've misunderstood something.

**Reply by yanns** [reply]
ah yes you're right. This is a new behavior, but different from the one I've originally described.
Before, those kind of updates were not opened as PR.
As we're using:
```
  "lockFileMaintenance": {
    "enabled": true
  }
```
the lock file was updated once per week, avoiding having lots of individual PRs.
[doc](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance)

Should I open a new discussion for this?

**Reply by rarkins** [reply]
No, read the docs on rangeStrategy and then configure it to "replace"

**Reply by rarkins** [reply]
Lock file maintenance is an inferior solution to what you're getting now though, because you won't know what's updated in each PR. Better to group and schedule the current PRs.

**Reply by yanns** [reply]
> Lock file maintenance is an inferior solution to what you're getting now though, because you won't know what's updated in each PR. Better to group and schedule the current PRs.

Yes it has its advantages. Thanks for the info.
